src/avram/add_column_attributes.cr

```diff
@@ -0,0 +1,125 @@
+          {% else %}
+            set_{{ attribute[:name] }}_from_param(value.as(String)) if key == {{ attribute[:name].stringify }}
+          {% end %}
+        {% end %}
+      end
+    end
+
+    def permitted_params
+      single_values = @params.nested(self.class.param_key).reject {|k,v| k.ends_with?("[]")}
+      array_values = @params.nested_arrays?(self.class.param_key) || {} of String => Array(String)
+      new_params = single_values.merge(array_values)
+      new_params.select(@@permitted_param_keys)
+    end
+
+    {% for attribute in attributes %}
+      @_{{ attribute[:name] }} : Avram::Attribute({{ attribute[:type] }})?
+
+      def {{ attribute[:name] }}
+        _{{ attribute[:name] }}
+      end
+
+      def {{ attribute[:name] }}=(_value)
+        \{% raise <<-ERROR
+          Can't set an attribute value with '{{attribute[:name]}} = '
+
+          Try this...
+
+            ▸ Use '.value' to set the value: '{{attribute[:name]}}.value = '
+
+          ERROR
+          %}
+      end
+
+      private def _{{ attribute[:name] }}
+        record_value = @record.try(&.{{ attribute[:name] }})
+        value = record_value.nil? ? default_value_for_{{ attribute[:name] }} : record_value
+
+        @_{{ attribute[:name] }} ||= Avram::Attribute({{ attribute[:type] }}).new(
+          name: :{{ attribute[:name].id }},
+          param: permitted_params["{{ attribute[:name] }}"]?,
+          value: value,
+          param_key: self.class.param_key)
+      end
+
+      private def default_value_for_{{ attribute[:name] }}
+        {% if attribute[:value] || attribute[:value] == false %}
+          parse_result = {{ attribute[:type] }}.adapter.parse({{ attribute[:value] }})
+          if parse_result.is_a? Avram::Type::SuccessfulCast
+            parse_result.value.as({{ attribute[:type] }})
+          else
+            nil
+          end
+        {% else %}
+          nil
+        {% end %}
+      end
+
+      {% if attribute[:type].is_a?(Generic) %}
+      def set_{{ attribute[:name] }}_from_param(_value : Array(String))
+        parse_result = {{ attribute[:type] }}.adapter.parse(_value)
+
+        if parse_result.is_a? Avram::Type::SuccessfulCast
+          {{ attribute[:name] }}.value = parse_result.value.as({{ attribute[:type] }})
+        else
+          {{ attribute[:name] }}.add_error "is invalid"
+        end
+      end
+      {% else %}
+      def set_{{ attribute[:name] }}_from_param(_value)
+        # In nilable types, `nil` is ok, and non-nilable types we will get the
+        # "is required" error.
+        if _value.blank?
+          {{ attribute[:name] }}.value = nil
+          return
+        end
+
+        parse_result = {{ attribute[:type] }}.adapter.parse(_value)
+
+        if parse_result.is_a? Avram::Type::SuccessfulCast
+          {{ attribute[:name] }}.value = parse_result.value.as({{ attribute[:type] }})
+        else
+          {{ attribute[:name] }}.add_error "is invalid"
+        end
+      end
```

Comment:
This was just 1 method before that did a quick check if the type was Generic or not before parsing. However, I think there was a bug doing it that way. We call `if _value.blank?` because if it's an empty string, treat it as `nil`. But in the case of an empty array, that could be a very valid value, and shouldn't be treated as nil.

Now, if we still want that same functionality (treating `[]` as `nil`), then I can add that back in, but I think that may have been a mistake.